packages/crayons-core/src/utils/draggable.ts

```diff
@@ -0,0 +1,302 @@
+import { debounce, cloneNodeWithEvents } from './index';
+
+//Global Variables
+let dragElement;
+const DEFAULT_OPTIONS = {
+  sortable: false,
+  acceptFrom: '',
+  placeholderClass: '',
+  copy: true,
+  addOnDrop: true,
+};
+export class Draggable {
+  host: HTMLElement;
+  options;
+  childElements = [];
+  acceptFrom = [];
+  placeholder: HTMLElement;
+  nextSibling: HTMLElement;
+  previousContainer;
+  cancelDebouncedDrag;
+
+  debouncedSetElement = debounce(
+    (childElements, draggingElement, y) => {
+      if (this.cancelDebouncedDrag) {
+        return;
+      }
+
+      const afterElement = this.getDragAfterElement(childElements, y);
+
+      let newElement;
+      // dragging inside the same container, so no need to add placeholder
+      if (
+        draggingElement.parentElement.id === this.host.id ||
+        !this.options.copy
+      ) {
+        newElement = draggingElement;
+      } else {
+        this.placeholder =
```

Comment:
`this.placeholder ||= this.createPlaceholder(draggingElement);`

Reply:
Nice one!! Thanks for the suggestion.